actions/push-to-gcs/action.yaml

```diff
@@ -0,0 +1,61 @@
+name: Push to cloud storage
+description: Composite action to push to Google Cloud Storage
+inputs:
+  object:
+    description: |
+      Name of object to upload.
+    required: true
+  destination:
+    description: |
+      Location to upload object (including object name). Will create sub-folders in the bucket.
+      Default behavior is to upload ${inputs.object} to the root folder of the bucket.
+    default: ""
+  environment:
+    description: |
+      Environment for uploading objects (can be either dev or prod).
+    default: dev
+
+runs:
+  using: composite
+  steps:
+    - name: Resolve GCP project
+      id: resolve-project
+      shell: bash
+      run: |
+        if [ "${{ inputs.environment }}" == 'dev' ]; then
+          PROJECT="grafanalabs-dev"
+        elif [ "${{ inputs.environment }}" == 'prod' ]; then
+          PROJECT="grafanalabs-global"
+        else
+          echo "Invalid environment. Valid environment variable inputs: dev, prod"
+          exit 1
+        fi
+        echo "project=${PROJECT}" >> ${GITHUB_OUTPUT}
+    - name: Get repository name
+      id: get-repository-name
+      shell: bash
+      run: |
+        REPO_NAME=$(echo ${{ github.repository }} | awk -F'/' '{print $2}')
+        echo "repo_name=${REPO_NAME}" >> ${GITHUB_OUTPUT}
+    - name: Get object name and location
+      id: get-object-location
+      shell: sh
+      run: echo "object_path=${{ inputs.destination != '' && inputs.destination || inputs.object }}" >> ${GITHUB_OUTPUT}
+    - name: Construct service account
+      id: construct-service-account
+      shell: sh
+      run: |
+        SERVICE_ACCOUNT="github-${{ steps.get-repository-name.outputs.repo_name }}-${{ inputs.environment }}@grafanalabs-workload-identity.iam.gserviceaccount.com"
```

Comment:
are service accounts being automatically generated for each repo or is that up to the codeowners to create them in deployment tools?

Reply:
the latter --- we have a TF module internally for that